Source/WebCore/rendering/mac/RenderThemeMac.mm

```diff
@@ -825,6 +859,222 @@ static NSControlSize controlSizeFromPixelSize(std::span<const IntSize, 4> sizes,
+{
+    static const std::array<IntSize, 4> sizes = { { IntSize(0, 20), IntSize(0, 16), IntSize(0, 13), IntSize(0, 28) } };
+    return sizes;
+}
+
+static std::span<const int, 4> buttonMargins(NSControlSize controlSize)
+{
+    // FIXME: These values may need to be reevaluated. They appear to have been originally chosen
+    // to reflect the size of shadows around native form controls on macOS, but as of macOS 10.15,
+    // these margins extend well past the boundaries of a native button cell's shadows.
+    static constexpr std::array margins {
+        std::array { 5, 7, 7, 7 },
+        std::array { 4, 6, 7, 6 },
+        std::array { 1, 2, 2, 2 },
+        std::array { 6, 6, 6, 6 },
+    };
+    return margins[controlSize];
+}
+
+// Stepper
+
+static const std::array<IntSize, 4>& stepperSizes()
+{
+    static const std::array<IntSize, 4> sizes = { { IntSize(19, 27), IntSize(15, 22), IntSize(13, 15), IntSize(19, 27) } };
+    return sizes;
+}
+
+// We don't use controlSizeForFont() for steppers because the stepper height
+// should be equal to or less than the corresponding text field height,
+static NSControlSize stepperControlSizeForFont(const FontCascade& font)
+{
+    auto fontSize = font.size();
+    if (fontSize >= 23 && ThemeMac::supportsLargeFormControls())
+        return NSControlSizeLarge;
+    if (fontSize >= 18)
+        return NSControlSizeRegular;
+    if (fontSize >= 13)
+        return NSControlSizeSmall;
+    return NSControlSizeMini;
+}
+
+// Switch
+
+static const std::array<IntSize, 4>& switchSizes()
+{
+    static const std::array<IntSize, 4> sizes =
+    {
+        IntSize { 38, 22 },
+        IntSize { 32, 18 },
+        IntSize { 26, 15 },
+        IntSize { 38, 22 }
+    };
+    return sizes;
+}
+
+static std::span<const int, 4> switchMargins(NSControlSize controlSize)
+{
+    static constexpr std::array margins {
+        // top right bottom left
+        std::array { 2, 2, 1, 2 },
+        std::array { 2, 2, 1, 2 },
+        std::array { 1, 1, 0, 1 },
+        std::array { 2, 2, 1, 2 },
+    };
+    return margins[controlSize];
+}
+
+static std::span<const int, 4> visualSwitchMargins(NSControlSize controlSize, bool isVertical)
+{
+    auto margins = switchMargins(controlSize);
+    if (isVertical) {
+        static const std::array verticalMargins { margins[3], margins[0], margins[1], margins[2] };
+        return verticalMargins;
+    }
+    return margins;
+}
+
+static LengthSize switchSize(const LengthSize& zoomedSize, float zoomFactor)
+{
+    // If the width and height are both specified, then we have nothing to do.
+    if (!zoomedSize.width.isIntrinsicOrAuto() && !zoomedSize.height.isIntrinsicOrAuto())
+        return zoomedSize;
+
+    return sizeFromNSControlSize(NSControlSizeSmall, zoomedSize, zoomFactor, switchSizes());
+}
+
+static void inflateControlPaintRect(StyleAppearance appearance, FloatRect& zoomedRect, float zoomFactor, bool isVertical)
+{
+    BEGIN_BLOCK_OBJC_EXCEPTIONS
```

Comment:
Where’s the ObjC?

Reply:
There is a little below various calls to `supportsLargeFormControls()`, but I kinda doubt that would ever throw. But given this is just a move, I am hesitant to remove it. Someone with AppKit source could probably double check it.